test/specs/errors/tests/07.http_409_error.js

```diff
@@ -137,15 +152,42 @@ exports.execute = function (options) {
            });
        });
 
-        it("if it's a duplicate key error, we should generate a more readable message.", function (done) {
-            nock(url, ops)
-                .get("/ermrest/catalog/1235")
-                .reply(409, duplicateErrorServerResponse)
-                .persist();
+        it("on create, if it's a duplicate key error, we should generate a more readable message.", function (done) {
+            duplicateReference.create([{duplicate_id: 1}]).then(null, function (err) {
+                expect(err.code).toBe(409, "invalid error code");
+                expect(err.message).toBe(duplicateErrorMappedMessage, "invalid error message");
+                expect(err.errorData.reference).toBeDefined();
+                done();
+            }).catch(function(err) {
+                console.log(err);
+                done.fail();
+            });
+        });
 
-            server.catalogs.get("1235").then(null, function(err) {
+        it("on update, if it's a duplicate key error, we should generate a more readable message.", function (done) {
+            duplicateReferenceUpdate.read(1).then(function (response) {
+                var data = response.tuples[0].data;
+
+                // updateData = { "duplicate_id": 2 }
+                data.duplicate_id = 2
+
+                return duplicateReferenceUpdate.update(response.tuples);
+            }).then(null, function (err) {
                 expect(err.code).toBe(409, "invalid error code");
                 expect(err.message).toBe(duplicateErrorMappedMessage, "invalid error message");
+                expect(err.errorData.reference).toBeDefined();
```

Comment:
Instead of just checking for existence of reference, check for its path (reference.location.compactPath) to make sure we're extracting the correct column name and values.